src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -453,7 +453,8 @@ private void Project(IDataView trainingData, float[][] mean, float[][][] omega,
                     activeColumns[sInfo.WeightColumnIndex] = true;
             }
 
-            using (var cursor = trainingData.GetRowCursor(col => activeColumns[col]))
+            var inputCols = trainingData.Schema.Where(x => activeColumns[x.Index]);
```

Comment:
>activeColumns [](start = 59, length = 13)

Is it possible to replace activeColumns with ColumnPairs->Input + WeightColumnIndex? Or worth it?

Reply:
looking into. I'll update in another iteration. 

---
In reply to: [246938070](https://github.com/dotnet/machinelearning/pull/1975#discussion_r246938070) [](ancestors = 246938070)

Reply:
I deem it not worth it. To figure out which of the  trainingData.Schema column, we'd still need to iterate/LINQ over all the sInfo and select all indexes, find those indexes in the trainingData.Schema columns. 

Feels similar, as far as complexity, to me. 

---
In reply to: [247287016](https://github.com/dotnet/machinelearning/pull/1975#discussion_r247287016) [](ancestors = 247287016,246938070)